frontend/src/components/Form/hooks/useForm.tsx

```diff
@@ -0,0 +1,51 @@
+import { h } from 'preact';
+import {
+  useState, useCallback, useEffect,
+} from 'preact/hooks';
+import { ValidationUtil } from '../../../utils/validationUtils';
+
+export const useForm = ({
+  validations,
+  initialValues,
+  onSubmit,
+}) => {
+  const [values, setValues] = useState(initialValues || {});
+  const [errors, setErrors] = useState({});
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [currentValue, setCurrentValue] = useState('');
+
+  const handleChange = useCallback((event) => {
+    const { target } = event;
+    const { name, value, defaultValue } = target;
+    if (defaultValue) setCurrentValue(value);
+
+    setValues((currentValues) => ({ ...currentValues, [name]: value }));
+    setErrors((currentErrors) => ({
+      ...currentErrors,
+      [name]: ValidationUtil.validateField(name, value, validations),
+    }));
+  }, []);
+
+  const handleSubmit = useCallback((event) => {
+    event.preventDefault();
+    const errorCheck = ValidationUtil.validate(values, validations);
+    setErrors(errorCheck);
+    setIsSubmitted(true);
+    if (isSubmitted) {
```

Comment:
remove this block

Reply:
If remove this block, the validation on the submission dosnt work.